src/services/TaskRepository.ts

```diff
@@ -1,83 +1,86 @@
-import { QueryTypes } from "sequelize";
+import { v4 as uuidv4 } from 'uuid';
+const Task = require('../models/SequelizeModels');
 
-const instances = require('hapi-sequelizejs').instances;
 export class TaskRepo {
-    
-    async getTasks(completion:string, sortBy:string, sortOrder:string){
-        var queryString = "SELECT * FROM Tasks ";
 
-        if (completion != null && completion != undefined)
-        {
-            queryString += `WHERE completed = ${completion} `;
+    async getTasks(completion:string, sortBy:string = 'dueDate', sortOrder:string = 'asc'){
 
-        }
+        const query = Task.TaskModel()
+        var results
 
-        if (sortBy != null && sortBy != undefined)
-        {
-            queryString += `ORDER BY ${sortBy} ${sortOrder}`;
+        // if completion parameter isn't given, just give all tasks.
+        if (completion != null || completion != undefined){
+            results = query.findAll({
+                where: {
+                    completed : JSON.parse(completion),
+                },
+                order: [
+                    [sortBy , sortOrder]
+                ]
+            })
+        }else{
+            results = query.findAll({
+                order: [
+                    [sortBy , sortOrder]
+                ]
+            })
         }
-        
-        var [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            queryString
-        )
-        return results;
+
+        return results
     }
 
-    async fetchTask(id:number) {
-        const [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            'SELECT * FROM Tasks WHERE id = ?',
-            {
-                replacements: [id],
-                type: QueryTypes.SELECT
+    async fetchTask(fetchID:number) {
+        const query = Task.TaskModel()
+
+        const results = query.findAll({
+            where: {
+                id: fetchID
             }
-        )
-        return results;
+        })
+
+        return results
     }
 
     async createTask(desc:string, due:Date, completed: boolean){
-        const newTask:ITask = {
-            id: 0,
+
+        const query = Task.TaskModel()
+
+        const newTask = await query.create({
+            id: uuidv4(),
             taskDescription:desc, 
-            createdAt: new Date(),
```

Comment:
I'm guessing you're using the createdAt from timestamps, correct? Are you going to be implementing updatedAt?

Reply:
I'm choosing not to, as it was not requested in the spec.